test/unit/sessions/job_entities/test_parameters_from_api_response.py

```diff
@@ -0,0 +1,130 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from typing import Union
+
+from openjd.model import ParameterValueType
+
+from deadline_worker_agent.sessions.job_entities.job_details import parameters_from_api_response
+from deadline_worker_agent.api_models import (
+    StringParameter,
+    PathParameter,
+    IntParameter,
+    FloatParameter,
+    ChunkIntParameter,
+)
+
+
+class TestParametersFromApiResponse:
+    def test_parameters_from_api_response_with_chunk_int(self):
+        # GIVEN - One parameter is chunked
+        params: dict[
+            str,
+            Union[
+                StringParameter, PathParameter, IntParameter, FloatParameter, ChunkIntParameter, str
+            ],
+        ] = {
+            "stringParam": {"string": "value"},
+            "pathParam": {"path": "/path/to/file"},
+            "intParam": {"int": "42"},
+            "floatParam": {"float": "3.14"},
+            "chunkIntParam": {"chunkInt": "1-5"},
+        }
+
+        # WHEN
+        result = parameters_from_api_response(params)
+
+        # THEN
+        assert len(result) == 5
+        assert result["stringParam"].type == ParameterValueType.STRING
+        assert result["stringParam"].value == "value"
+        assert result["pathParam"].type == ParameterValueType.PATH
+        assert result["pathParam"].value == "/path/to/file"
+        assert result["intParam"].type == ParameterValueType.INT
+        assert result["intParam"].value == "42"
+        assert result["floatParam"].type == ParameterValueType.FLOAT
+        assert result["floatParam"].value == "3.14"
+        assert result["chunkIntParam"].type == ParameterValueType.CHUNK_INT
+        assert result["chunkIntParam"].value == "1-5"
+
+    def test_parameters_from_api_response_with_contiguous_chunks(self):
+        # GIVEN - Different formats of contiguous chunk ranges
+
+        # Test case 1: Standard range format
+        params1: dict[
+            str,
+            Union[
+                StringParameter, PathParameter, IntParameter, FloatParameter, ChunkIntParameter, str
+            ],
+        ] = {"chunkIntParam": {"chunkInt": "1-5"}}
+        result1 = parameters_from_api_response(params1)
+        assert result1["chunkIntParam"].type == ParameterValueType.CHUNK_INT
+        assert result1["chunkIntParam"].value == "1-5"
+
+        # Test case 2: Single frame case (should be expressed as a range with same start and end)
+        params2: dict[
+            str,
+            Union[
+                StringParameter, PathParameter, IntParameter, FloatParameter, ChunkIntParameter, str
+            ],
+        ] = {"chunkIntParam": {"chunkInt": "100-100"}}
+        result2 = parameters_from_api_response(params2)
+        assert result2["chunkIntParam"].type == ParameterValueType.CHUNK_INT
+        assert result2["chunkIntParam"].value == "100-100"
```

Comment:
Another nit, since this is a separate test case it should be it's own test, or parametrized using `pytest.mark.parametrize`. Use can use the `id` kwarg in order to label specific cases like this.  Here's an example in this repo of what that looks in the [temporary credentials tests](https://github.com/aws-deadline/deadline-cloud-worker-agent/blob/4c126648a657df83e065c77274c965b8d0ad283e/test/unit/aws_credentials/test_temporary_credentials.py#L310-L318).